test.disabled/ejabberd_tests/tests/muc_SUITE.erl

```diff
@@ -432,6 +504,47 @@ init_per_testcase(CaseName, ConfigIn) ->
     Config = maybe_create_unique_room(ConfigIn),
     escalus:init_per_testcase(CaseName, Config).
 
+%% Meck will register a fake room right before a 'real' room is started
+meck_room_start() ->
```

Comment:
There is one issue with this approach. It's very implementation specific. For example: If some would move `mod_muc_room:start` call before the room check (which btw. would be totally fine), those tests stop testing anything. This is only a note, since I don't really think other ideas I have are objectively better (e.g. we could slow down `mnesia:write` significantly, to like 5 seconds, which would make two requests always create race condition). Just something to think about :)